### ISSUE TITLE
Error during istio-proxy startup

### ISSUE THREAD

**Issue by Stono** [istio/istio#39546]
### Bug Description

Hi, 
I run Istio on GKE, but during startup we're seeing:

```
{"level":"error","time":"2022-06-22T07:27:08.870922Z","msg":"error in getting aws info for iam/info : unexpected status 404"}
```

I assume this call to get aws iam info shouldn't be happening on GKE, or it shouldn't be logged @ level `error`.

### Version

```prose
`1.13.5` on GKE
```

### Affected product area

- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [X] User Experience
- [X] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### Is this the right place to submit this?

- [X] This is not a security vulnerability
- [X] This is not a question about how to use Istio

**Comment by Jeskz0rd** [comment]
I could also see it on 1.13.4 also on GKE.

**Comment by howardjohn** [comment]
afaik this is fixed in 1.14
